internal/vault/secrets/ephemeral/gcp_oauth2_access_token.go

```diff
@@ -0,0 +1,321 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package ephemeralsecrets
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/base"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/client"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/errutil"
+	"github.com/hashicorp/vault/api"
+)
+
+// Ensure the implementation satisfies the ephemeral.EphemeralResource interface
+var _ ephemeral.EphemeralResource = &GCPOAuth2AccessTokenEphemeralResource{}
+
+// NewGCPOAuth2AccessTokenEphemeralResource returns the implementation for this resource to be
+// imported by the Terraform Plugin Framework provider
+var NewGCPOAuth2AccessTokenEphemeralResource = func() ephemeral.EphemeralResource {
+	return &GCPOAuth2AccessTokenEphemeralResource{}
+}
+
+// GCPOAuth2AccessTokenEphemeralResource implements the methods that define this resource
+type GCPOAuth2AccessTokenEphemeralResource struct {
+	base.EphemeralResourceWithConfigure
+}
+
+// GCPOAuth2AccessTokenModel describes the Terraform resource data model to match the
+// resource schema.
+type GCPOAuth2AccessTokenModel struct {
+	// common fields to all ephemeral resources
+	base.BaseModelEphemeral
+
+	// fields specific to this resource
+	Backend             types.String `tfsdk:"backend"`
+	Roleset             types.String `tfsdk:"roleset"`
+	StaticAccount       types.String `tfsdk:"static_account"`
+	ImpersonatedAccount types.String `tfsdk:"impersonated_account"`
+
+	// computed fields
+	Token          types.String `tfsdk:"token"`
+	TokenTTL       types.Int64  `tfsdk:"token_ttl"`
+	LeaseID        types.String `tfsdk:"lease_id"`
+	LeaseDuration  types.Int64  `tfsdk:"lease_duration"`
+	LeaseStartTime types.String `tfsdk:"lease_start_time"`
+	LeaseRenewable types.Bool   `tfsdk:"lease_renewable"`
+}
+
+// Schema defines this resource's schema which is the data that is available in
+// the resource's configuration, plan, and state
+func (r *GCPOAuth2AccessTokenEphemeralResource) Schema(_ context.Context, _ ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"backend": schema.StringAttribute{
+				MarkdownDescription: "GCP Secret Backend to read credentials from.",
+				Required:            true,
+			},
+			"roleset": schema.StringAttribute{
```

Comment:
nit: are we sure that shouldn't be spelled `role_set` - it seems like two words.